### DISCUSSION TITLE
can decode api to extract gain map metadata only?

### DISCUSSION THREAD

**Discussion by GrousexyHKCN** [google/libultrahdr#190] [Q&A] [answered]
is it possible to extract the gain map metadata then save to file?

**Comment by ram-mohan** [comment]
You can use uhdr_dec_get_gain_map_metadata() to get metadata. The return value is of type uhdr_gainmap_metadata_t. You can write the fields to a file.

**Comment by pmonck** [comment]
Could this functionality be added to the demo app? It would be great to be able to extract, edit and copy metadata between ultrahdr jpegs.

**Comment by ram-mohan** [comment]
will try to add a pull request to handle this. Thank you, 
For urgent purposes, Below code should work,

`uhdr_codec_private_t* handle = uhdr_create_decoder();`
`RET_IF_ERR(uhdr_dec_set_image(handle, &mUhdrImage))`
`RET_IF_ERR(uhdr_dec_probe(handle))  // RET_IF_ERR(uhdr_decode(handle)) also works but for just metadata probe is enough`
`uhdr_gainmap_metadata_t* metadata = uhdr_dec_get_gain_map_metadata(handle);`
`// write fields of metadata descriptor to a file.`
`uhdr_release_decoder(handle);`

**Comment by ram-mohan** [comment, marked as answer]
@pmonck @GrousexyHKCN the demo app contains the changes that supports this.
